cpp/bench/sg/arima_loglikelihood.cu

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuml/cuml.hpp>
+
+#include <thrust/execution_policy.h>
+#include <thrust/for_each.h>
+#include <thrust/iterator/counting_iterator.h>
+
+#include <cuml/tsa/arima_common.h>
+#include <random/rng.h>
+#include <cuml/tsa/batched_arima.hpp>
+
+#include "benchmark.cuh"
+
+namespace ML {
+namespace Bench {
+namespace Arima {
+
+struct ArimaParams {
+  TimeSeriesParams data;
+  ARIMAOrder order;
+};
+
+template <typename D>
+class ArimaLoglikelihood : public TsFixtureRandom<D> {
+ public:
+  ArimaLoglikelihood(const std::string& name, const ArimaParams& p)
+    : TsFixtureRandom<D>(p.data), order(p.order) {
+    this->SetName(name.c_str());
+  }
+
+  // Note: public function because of the __device__ lambda
+  void runBenchmark(::benchmark::State& state) override {
+    auto& handle = *this->handle;
+    auto stream = handle.getStream();
+    auto allocator = handle.getDeviceAllocator();
+    auto counting = thrust::make_counting_iterator(0);
+
+    // Generate random parameters
+    int N = order.complexity();
+    D* x =
+      (D*)allocator->allocate(N * this->params.batch_size * sizeof(D), stream);
+    MLCommon::Random::Rng gpu_gen(this->params.seed,
```

Comment:
How about using the `gpu_gen` that's in `TimeSeriesDataset`? That way we don't have to maintain 2 different seeds.

Reply:
They are already using the same seed.
I could reuse the `Rng` object but I'm not sure it's worth the trouble

Reply:
ok.. let's leave it at this for now.